src/popup/router/components/UserAvatar.vue

```diff
@@ -1,7 +1,10 @@
 <template>
   <img
-    v-if="!error"
-    :src="typeof profileImage === 'string' && profileImage.length > 0 ? profileImage : avatar.src"
+    :src="
+      typeof profileImage === 'string' && profileImage.length > 0 && !error
+        ? profileImage
+        : avatar.src
+    "
```

Comment:
Can't we use https://github.com/aeternity/superhero-avatars for this? There is an instance up and running on https://avatars.z52da5wt.xyz/.

Check it out: `https://avatars.z52da5wt.xyz/{CHAIN_NAME || AK_ADDRESS}`

Reply:
Thanks for sharing. I would rather use the libraries we have included though. In my opinion it will serve better as a fallback if the call for profileImage fails for some reason.

Reply:
Was it needed to refactor the avatars component in this PR? Actually, I like to extract avatars to a service, because packages for generation are too heavy.

Reply:
I just combined two <img> and one <div> tags into one <img>. The needed refactoring was regarding the call for profileImage. My proposition is not to overcomplicate the already big PR.

Reply:
https://github.com/aeternity/superhero-wallet/issues/551